dartagnan/src/main/java/com/dat3m/dartagnan/program/arch/linux/event/cond/FenceCond.java

```diff
@@ -48,6 +49,6 @@ protected BooleanFormula encodeExec(SolverContext ctx){
 
     @Override
     protected RecursiveAction unrollRecursive(int bound, Event predecessor, int depth) {
-        throw new RuntimeException("FenceCond cannot be unrolled: event must be generated during compilation");
+        throw new UnrollException("FenceCond cannot be unrolled: event must be generated during compilation");
```

Comment:
`UnsupportedOperationException` should do the trick. Possibly a custom `ProgramProcessingError`.